overlord/devicestate/devicestate.go

```diff
@@ -235,6 +236,7 @@ func delayedCrossMgrInit() {
 	snapstate.CanAutoRefresh = canAutoRefresh
 	snapstate.CanManageRefreshes = CanManageRefreshes
 	snapstate.IsOnMeteredConnection = netutil.IsOnMeteredConnection
+	snapstate.IsNTPSynchronized = timeutil.IsNTPSynchronized
```

Comment:
instead of adding a new func hook I think it would make more sense to incorporate the new logic into CanAutoRefresh, no? so we cover also refreshhints btw.  It's just a bit harder to keep state but there's a couple of ways to do that

Reply:
Thanks, this is exactly the kind of advise I had hoped for, happy to add it there. Thank you!